malariagen_data/anoph/snp_data.py

```diff
@@ -1673,20 +1683,21 @@ def biallelic_snp_calls(
 
                 # Apply minor allele count condition.
                 if min_minor_ac is not None:
-                    ac_minor = ac_out.min(axis=1)
+                    ac_minor = ac_out_computed.min(axis=1)
                     if isinstance(min_minor_ac, float):
                         ac_minor_frac = ac_minor / an
                         loc_minor = ac_minor_frac >= min_minor_ac
                     else:
                         loc_minor = ac_minor >= min_minor_ac
                     loc_out &= loc_minor
 
-                ds_out = ds_out.isel(variants=loc_out)
+                # Apply selection from conditions.
+                ds_out = dask_compress_dataset(ds_out, indexer=loc_out, dim="variants")
```

Comment:
This is the main optimisation. Our implementation of `dask_compress_dataset()` is much faster than calling `ds.isel()`.